addons/website/static/src/js/content/menu.js

```diff
@@ -660,6 +657,130 @@ publicWidget.registry.hoverableDropdown = animations.Animation.extend({
+    disabledInEditableMode: false,
+    events: {
+        "mousedown .o_mega_menu_toggle": "_onMegaMenuClick",
+        "mouseenter .o_mega_menu_toggle": "_onMegaMenuHover",
+        "mousedown .o_extra_menu_items": "_onExtraMenuClick",
+    },
+
+    /**
+     * @override
+     */
+    start() {
+        this.desktopMegaMenuToggleEls = [...this.el.querySelectorAll(".navbar:not(.o_header_mobile) .o_mega_menu_toggle")];
+        this.mobileMegaMenuToggleEls = [...this.el.querySelectorAll(".navbar.o_header_mobile .o_mega_menu_toggle")];
+        // If the mega menus are duplicated (i.e. they are in both desktop and
+        // mobile view navbars), only keep one of them. Indeed, having the same
+        // mega menu multiple times can cause some issues when editing.
+        // Note: The XML templates have been modified to add them only once in
+        // the DOM. This code is needed to fix the issues for databases created
+        // before the templates modifications.
+        if (this.el.querySelectorAll(".o_mega_menu").length === 2 * this.desktopMegaMenuToggleEls.length) {
+            this.desktopMegaMenuToggleEls.forEach((megaMenuToggleEl, i) => {
+                const hasMegaMenu = !!megaMenuToggleEl.parentElement.querySelector(".o_mega_menu");
+                if (hasMegaMenu) {
+                    const mobileMegaMenuEl = this.mobileMegaMenuToggleEls[i].parentElement.querySelector(".o_mega_menu");
+                    if (mobileMegaMenuEl) {
+                        mobileMegaMenuEl.remove();
+                    }
+                }
+            });
+        }
+
+        return this._super(...arguments);
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * If the mega menu dropdown on which we are clicking/hovering does not have
+     * a mega menu (i.e. it is in the other navbar), brings the corresponding
+     * mega menu into it.
+     *
+     * @private
+     * @param {Element} megaMenuToggleEl the clicked/hovered mega menu dropdown
+     */
+    _moveMegaMenu(megaMenuToggleEl) {
+        const hasMegaMenu = !!megaMenuToggleEl.parentElement.querySelector(".o_mega_menu");
+        if (hasMegaMenu) {
+            return;
+        }
+        this.options.wysiwyg && this.options.wysiwyg.odooEditor.observerUnactive("moveMegaMenu");
+        let megaMenuEl, previousDropdownEl;
+        const megaMenuId = megaMenuToggleEl.querySelector("span").dataset.oeId;
+        // If the `data-oe-id` attribute is there, we directly use it to find
+        // the corresponding mega menu (to be more efficient). Otherwise, we use
+        // the menus order.
+        if (megaMenuId) {
+            megaMenuEl = this.el.querySelector(`.o_mega_menu[data-oe-id="${megaMenuId}"]`);
+            previousDropdownEl = megaMenuEl.parentElement.querySelector(".o_mega_menu_toggle");
```

Comment:
by "previous" I mean the dropdown where the mega menu was before moving it (so the previous dropdown of the mega menu) 
I find it more clear than "hidden" (because "what is hidden ?" then) 
same for other/current, in this case, I find previous more relevant, as it is really linked to the mega menu
However, I could write `previousMegaMenuDrodownEl` to be more clear :wink:

Reply:
I understood what you meant, but only after thinking about it more than I should have. Maybe it's just me, but it's really the use of "previous" that wasn't clear to me. It's the "previously active dropdown" / "previous mega menu container dropdown", it's not the "previous dropdown".
That's why I think having a consistent naming convention across your function(s) to refer to the active/unactive dropdown would be clearer.